### ISSUE TITLE
New project fails at blt setup with AnnotationRegistry.php on line 50 error

### ISSUE THREAD

**Issue by markf3lton** [acquia/blt#2537]
My system information:
* Operating system type: macOS 
* Operating system version: 10.13.3
* BLT version: 8.9.12 

Output of `blt doctor`:
```
mark.felton$ blt doctor
Drupal VM was detected. Running blt doctor inside of VM...
Executing command cd /var/www/my-project/docroot/.. && /var/www/my-project/docroot/../vendor/bin/drush cc drush && /var/www/my-project/docroot/../vendor/bin/drush --include=/var/www/my-project/docroot/../vendor/acquia/blt/drush blt-doctor -r /var/www/my-project/docroot/../docroot inside of Drupal VM...
'drush' cache was cleared.                                                                                     [success]
+----------------------+----------------------------------------------------------------------------------+
| Property             | Value                                                                            |
+----------------------+----------------------------------------------------------------------------------+
| %paths               |                                                                                  |
|  - %root             | /var/www/my-project/docroot                                                      |
|  - %site             | sites/default                                                                    |
|  - %modules          | modules                                                                          |
|  - %themes           | themes                                                                           |
|  - %config-sync      | /var/www/my-project/config/default                                               |
| blt-version          | 8.9.12                                                                           |
| config-sync          | /var/www/my-project/config/default                                               |
| db-driver            | mysql                                                                            |
| db-hostname          | localhost                                                                        |
| db-name              | drupal                                                                           |
| db-password          | drupal                                                                           |
| db-port              | 3306                                                                             |
| db-username          | drupal                                                                           |
| dev-desktop-enabled  | false                                                                            |
| drupal-settings-file | sites/default/settings.php                                                       |
| drupal-version       | 8.4.4                                                                            |
| drupal-vm-enabled    | true                                                                             |
| drush-alias-files    | /var/www/my-project/docroot/../drush/site-aliases/example.local.aliases.drushrc. |
|                      | php                                                                              |
|                      | /var/www/my-project/docroot/../drush/site-aliases/aliases.drushrc.php            |
|                      | /var/www/my-project/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.p |
|                      | hp                                                                               |
| drush-conf           | /var/www/my-project/drush/drushrc.php                                            |
| drush-script         | /var/www/my-project/vendor/drush/drush/drush.php                                 |
| drush-temp           | /tmp                                                                             |
| drush-version        | 8.1.16                                                                           |
| local-drushrc        | /var/www/my-project/docroot/sites/default/local.drushrc.php                      |
| local-settings       | /var/www/my-project/docroot/sites/default/settings/local.settings.php            |
| modules              | modules                                                                          |
| php-bin              | /usr/bin/php                                                                     |
| php-conf             | /etc/php/5.6/cli/php.ini                                                         |
| php-mysql            | /var/run/mysqld/mysqld.sock                                                      |
| php-os               | Linux                                                                            |
| root                 | /var/www/my-project/docroot                                                      |
| shell                |                                                                                  |
| site                 | sites/default                                                                    |
| themes               | themes                                                                           |
| uri                  | http://local.my-project.com                                                      |
+----------------------+----------------------------------------------------------------------------------+
+--------------------------+----------------------------------------------------------------+
| Check                    | Outcome                                                        |
+--------------------------+----------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                 |
| checkCoreExists          | Drupal core exists                                             |
| checkSettingsFile        | BLT settings are included in settings file.                    |
| checkLocalSettingsFile   | Found your local settings file.                                |
| checkLocalDrushFile      | Found your local drush settings file.                          |
| checkUri                 | $options['uri'] is set.                                        |
| checkUriResponse         | Received a response from site http://local.my-project.com.     |
| checkFileSystem:%files   | Public files directory is not set.                             |
| checkFileSystem:%private | Private files directory is not set.                            |
| checkFileSystem:%temp    | Temporary files directory is not set.                          |
| checkDbConnection        | Connected to database.                                         |
| checkDrupalBootstrapped  | Could not bootstrap Drupal via drush without alias.            |
| checkDrupalInstalled     | Drupal is not installed.                                       |
|                          |                                                                |
|                          | Run `blt local:setup` to install Drupal locally.               |
| checkCaching:page        | Drupal cache is disabled.                                      |
| checkCaching:css         | CSS preprocessing is disabled.                                 |
| checkCaching:js          | JS preprocessing is disabled.                                  |
| checkComposer:require    | acquia/blt is in composer.json's require object.               |
| checkBehat:exists        | tests/behat/local.yml is missing!                              |
|                          |   Run `blt setup:behat` to generate it from example.local.yml. |
| checkProjectYml:keys     | project.yml has no deprecated keys.                            |
| checkDrupalVm:alias      | drush.aliases.local exists your drush aliases file.            |
| checkDrupalVm            | Drupal VM is configured correctly.                             |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                 |
+--------------------------+----------------------------------------------------------------+
Connection to 127.0.0.1 closed.
```

When I run this command:
```
composer create-project --no-interaction acquia/blt-project my-project 
cd my-project
blt vm
blt setup
```

I get the following output when I visit local.my-project.com in a browser:
```
Parse error: syntax error, unexpected ':', expecting ';' or '{' in /var/www/my-project/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationRegistry.php on line 50
```

And I expected this to happen:
I expected to see the Drupal setup screen.

I fixed it by changing "5.6" to "7.1" in box/config.yml and then running `vagrant reload --provision`

I suspected it might be a PHP version issue because of this post on [StackOverflow](https://stackoverflow.com/questions/45433204/symfony-2-8-parse-error-syntax-error-unexpected-expecting-or-in-c)

Before filing this issue, I ran the above sequence on two different computers and go the same result.

**Comment by mikemadison13** [comment]
what version of PHP on your host machine?

sounds like a php mismatch for sure.

**Comment by markf3lton** [comment]
I had previously used homebrew to install php, so

```
mark.felton$ php --version
PHP 7.1.14 (cli) (built: Feb  2 2018 08:42:59) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
```

Could that cause this? On one of my othermachines (macOS Sierra) I will remove the homebrew PHP and revert to the Apple default PHP which was 5.6 and try this again, to confirm this.

**Comment by mikemadison13** [comment]
i'm not sure that it matters if you are running 7.1.x on both the host and VM machines. it might make sense to vagrant ssh onto the box and do a php -v though just to make sure they are both 7.1.x?

**Comment by mikemadison13** [comment]
also maybe re-run blt doctor and paste the results now that you've re-provisioned the vm? the above still shows php 5.6 on the output.

**Comment by markf3lton** [comment]
Confirmed that I receive no errors creating new projects if my host machine and drupalvm use the same version of php. 

In my case, I originally and "5.6" in my box/config.yml but I was using php71 on my Mac. The easiest fix was to put "7.1" into box/config.yml and then run `vagrant reload --provision`.

Another option was to remove php71 and revert to php56, but more steps were involved, e.g.

```
brew unlink php71
brew install php56
brew services start homebrew/php/php56
```

When I changed php versions, I had to remember to update the php.ini settings file (located with `php -i | grep php.ini`) in order to increase the memory_limit to "2G". (I was prompted to set `date.timezone` as well.)

I believe a fresh install of macOS Sierra provides php56, but a fresh install of macOS _High_ Sierra ships with php71. My preference is to start new projects with php71 anyway.

Thanks for the support on this issue!

**Comment by markf3lton** [comment]
Hmm, when I run `$ php -i | grep php.ini` on a new Mac with High Sierra there is no default configuration file.

```
$ php --version
PHP 7.1.7 (cli) (built: Jul 15 2017 18:08:09) ( NTS )
Copyright (c) 1997-2017 The PHP Group
```

Then

```
$ php -i | grep php.ini
Configuration File (php.ini) Path => /etc
```

I can follow https://stackoverflow.com/a/9343210, but it was just as easy to

```
brew install php71
```

Then

```
$ php -i | grep php.ini
Configuration File (php.ini) Path => /usr/local/etc/php/7.1
Loaded Configuration File => /usr/local/etc/php/7.1/php.ini
```

Once this is done, remember to edit `usr/local/etc/php/7.1/php.ini` to set the default memory_limit of 128M (see line 404) to something like 2G.
